app/src/ui/lib/sandboxed-markdown.tsx

```diff
@@ -0,0 +1,222 @@
+   * nonce to run in the iframe.
+   */
+  private setContentSecurityPolicy(frameRef: HTMLIFrameElement): void {
+    const contentSecurityPolicy = `script-src 'nonce-${this.scriptNonce}'`
+    frameRef.setAttribute('csp', contentSecurityPolicy)
+  }
+
+  /**
+   * We still want to be able to navigate to links provided in the markdown.
+   * However, we want to intercept them an verify they are valid links first.
+   */
+  private setupLinkInterceptor(frameRef: HTMLIFrameElement): void {
+    frameRef.onload = () => {
+      if (frameRef.contentDocument === null) {
+        return
+      }
+      const linkTags = frameRef.contentDocument.querySelector('a')
+      if (linkTags === null) {
+        return
+      }
+
+      linkTags.addEventListener('click', (e: MouseEvent) => {
+        e.preventDefault()
+        // The point of intercepting them is to bubble them up.
+        // If no callback provided, no need to continue.
+        if (this.props.onMarkdownLinkClicked === undefined) {
+          return
+        }
+
+        if (e.target === null) {
+          log.warn('Failed to parse markdown link href')
+          return
+        }
+
+        let url
+        try {
+          const href = (e.target as HTMLAnchorElement).href
+          url = new URL(href)
+        } catch (_) {
+          log.warn('Failed to parse markdown link href')
+          return
+        }
+
+        if (url.protocol !== 'https:' && url.protocol !== 'http:') {
+          log.warn(
+            'Failed to parse markdown link href - non https: and http: links are blocked'
+          )
+          return
+        }
+
+        this.props.onMarkdownLinkClicked(url.toString())
+      })
+    }
+  }
+
+  /**
+   * Generates an iframe with a csp allowed script of a markdown parser.
+   */
+  private mountIframeContents = async (): Promise<void> => {
+    if (this.frameRef === null) {
+      return
+    }
+
+    const markedJS = await FSE.readFile(
+      Path.join(__dirname, 'static', 'marked.min.js'),
+      'utf8'
+    )
+
+    const styleSheet = await this.getInlineStyleSheet()
+    this.setContentSecurityPolicy(this.frameRef)
+
+    // We need to convert the markdown into a base64 string to allow us to print
+    // it into a script file and not break into multiple lines. Then, in the
+    // script file, we need to convert it back to a markdown string.
+    // encodeURIComponent/decodeURIComponent is important for things like emoji
+    // unicode and btoa fails to parse non latin strings.
+    const mardownToAndFromBase64 = `var md = decodeURIComponent(atob('${btoa(
+      encodeURIComponent(this.props.markdown)
+    )}'));`
+
+    // We want the marked parse to use GitHub Flavored Markdown (gfm)
+    const useGFM = `marked.use({
+                      gfm: true
+                    });`
+
+    const src = `
+      <html>
+      <head>
+        ${
+          this.props.baseHref !== null
+            ? `<base href="${this.props.baseHref}" />`
+            : ''
+        }
```

Comment:
This is likely not an actual risk since we're currently only using it for endpoints but since we might end up using this component for more purposes than this one I'd rather we properly escape strings we're mixing with the markup. How about something like this?

```suggestion
        ${base ? base.outerHTML : ''}
```

Coupled with

```ts
  let base: HTMLBaseElement | null = null
  if (this.props.baseHref !== null) {
    base = document.createElement('base', {})
    base.href = this.props.baseHref
  }
```